src/workerd/io/compatibility-date.c++

```diff
@@ -160,6 +169,28 @@ void compileCompatibilityFlags(kj::StringPtr compatDate, capnp::List<capnp::Text
         enableByDate = true;
       } else if (annotation.getId() == EXPERIMENTAl_ANNOTATION_ID) {
         isExperimental = true;
+      } else if (annotation.getId() == IMPLIED_BY_AFTER_DATE_ANNOTATION_ID) {
+        if (maybeImpliedBy == kj::none) {
+          auto value = annotation.getValue();
+          auto s = value.getStruct().getAs<workerd::ImpliedByAfterDate>();
+          auto parsedDate = KJ_ASSERT_NONNULL(CompatDate::parse(s.getDate()));
+          // This flag will be marked as enabled if the flag identified by
+          // s.getName() is enabled, but only on or after the specified date.
+          if (parsedCompatDate >= parsedDate && !disableByFlag) {
+            maybeImpliedBy.emplace(ImpliedBy {
+              .field = field,
+              .other = schema.getFieldByName(s.getName()),
+            });
+          }
+        }
+      }
+    }
+    KJ_IF_SOME(impliedBy, maybeImpliedBy) {
+      // We only want to add the implied by flag if it is not explicitly disabled.
+      // We also only want to add it if the flag is not experimental or experimental
+      // flags are allowed...
+      if (!disableByFlag && (!isExperimental || allowExperimentalFeatures)) {
```

Comment:
I think we should remove the `&& (!isExperimental || allowExperimentalFeatures)` part. Otherwise, interpretation of flags depends on whether or not the user is permitted to use experimental flags, which means Cloudflare employees will get differing behavior without realizing it.

If an experimental flag is turned on because it was implied, but experimental flags aren't allowed, I think it's fine that we just end up erroring.

In practice this probably won't ever matter since I don't think we'll ever define an experimental flag to be implied by a non-experimental flag anyway.